src/control/control.cpp

```diff
@@ -125,6 +130,32 @@ void ControlDoublePrivate::insertAlias(const ConfigKey& alias, const ConfigKey&
     s_qCOHash.insert(alias, pControl);
 }
 
+// static
+void ControlDoublePrivate::insertGroupAlias(
+        const QString& aliasGroup, const QString& originalGroup) {
+    MMutexLocker locker(&s_qCOHashMutex);
+
+    // Check if there are any pre-existing control with the group that we want
+    // to add an alias for and control aliases for them.
+    for (auto it = s_qCOHash.cbegin(), end = s_qCOHash.cend(); it != end; ++it) {
+        const auto originalKey = it.key();
+        if (originalKey.group != originalGroup) {
+            continue;
+        }
+
+        QSharedPointer<ControlDoublePrivate> pControl = it.value();
+        if (pControl.isNull()) {
+            continue;
+        }
+
+        const auto aliasKey = ConfigKey(aliasGroup, originalKey.item);
+        s_qCOAliasHash.insert(originalKey, aliasKey);
+        s_qCOHash.insert(aliasKey, pControl);
```

Comment:
That way, as part of the lookup we could also print a warning if the alias was used instead of the non-aliased group.

Reply:
> could this get expensive if the group is large?

You mean memory-wise? It adds a regular alias for every member in a group, but it shouldn't be that much of an issue. In terms of runtime performance, it shouldn't be an issue because lookups are only performed when a new CO is created and it's O(1).

> would it be better to do the aliasing for the group on lookup instead? That way, as part of the lookup we could also print a warning if the alias was used instead of the non-aliased group.

True, this could be worth considering. I'll have a look.

Reply:
> You mean memory-wise? It adds a regular alias for every member in a group, but it shouldn't be that much of an issue.

Right, if the group is large, this could bloat the table quite a bit. Yes, lookups are $O(1)$ but creating an a group alias and COs becomes $O(n)$ (where $n$ is size of the group in the former case and the number of alias groups in the latter). The increased table size would cause the usual slowdowns associated with high memory consumption (data cache misses, more yields to the OS for more memory, more table relocations since it grows faster / more often, etc). 

Doing the aliasing lookup would avoid that and only be worse performing on the unhappy path (probing via alias group, if CO exists return (happy path) otherwise lookup primary group name, probe again using primary group name, print warning, etc).